RecodoDesktop/Recodo.Desktop.Logic/DefaultBrowser.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Recodo.Desktop.Logic
+{
+    public class DefaultBrowser
+    {
+        private readonly string _url;
+        private readonly string _redirectUrl;
+        public DefaultBrowser(string url, string redirectUrl)
+        {
+            _url = url;
+            _redirectUrl = redirectUrl;
+        }
+
+        public async Task<string> InvokeAsync(CancellationToken cancellationToken = default)
+        {
+            using var listener = new LoopbackHttpListener(_redirectUrl);
+            Open(_url);
+
+            try
+            {
+                var result = await listener.WaitForCallbackAsync();
+                if (string.IsNullOrWhiteSpace(result))
+                {
+                    return "Empty response.";
+                }
+
+                return result;
+            }
+            catch (TaskCanceledException ex)
+            {
+                return ex.Message;
+            }
+            catch (Exception ex)
+            {
+                return ex.Message;
+            }
+        }
```

Comment:
Can we set a time to leave for this service? Or what will happen if user opens page, but will make nothing. And repeat this action 10 times? Will we listen to 10 ports?

Reply:
Yes it's open ten ports, and if user nothing to do listener wait when expired timeout. After timeout expired it throw exception. Timeout set in the listener.